packages/front-end/pages/api/init.ts

```diff
@@ -66,6 +68,7 @@ export default function handler(req: NextApiRequest, res: NextApiResponse) {
         : "enable",
     sentryDSN: NEXT_PUBLIC_SENTRY_DSN || "",
     usingSSO: !!SSO_CONFIG,
+    storeSegmentsInMongo: !!STORE_SEGMENTS_IN_MONGO,
```

Comment:
In `secrets.ts` on the back-end, you are doing `STORE_SEGMENTS_IN_MONGO === "true"`, here you are just making sure it's not empty.  We should be doing the exact same check in both places.  I like this version better (just check if it's not empty).